source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -120,8 +120,15 @@ Filter::StreamOpenState Filter::openStream() {
   ENVOY_BUG(!processing_complete_, "openStream should not have been called");
   if (!stream_) {
     ENVOY_LOG(debug, "Opening gRPC stream to external processor");
-    stream_ = client_->start(*this, grpc_service_, decoder_callbacks_->streamInfo());
-    stats_.streams_started_.inc();
+    if (grpc_service_.target_specifier_case() ==
+        envoy::config::core::v3::GrpcService::TargetSpecifierCase::TARGET_SPECIFIER_NOT_SET) {
+      // No valid gRPC service was specified and treated as an grpc error directly.
+      ENVOY_LOG(warn, "No gRPC service specified for external processing filter");
+      onGrpcError(Grpc::Status::WellKnownGrpcStatus::Unavailable);
```

Comment:
In this case, should we return StreamOpenState::Error here, or let it fall through to return  StreamOpenState::Ok?

Reply:
> It does look to me make the grpc_service in the message ExternalProcessor a required config will make the code cleaner

it's done now.